frame/support/src/storage/generator/double_map.rs

```diff
@@ -340,7 +340,9 @@ impl<
 	G::Hasher1: ReversibleStorageHasher,
 	G::Hasher2: ReversibleStorageHasher
 {
+	type KeyPrefixIterator = KeyPrefixIterator<K2>;
 	type PrefixIterator = PrefixIterator<(K2, V)>;
+	type KeyIterator = KeyPrefixIterator<(K1, K2)>;
```

Comment:
Perhaps these types could be `FirstKeyIterator` and `BothKeyIterator`? I can see what you mean by `KeyPrefixIterator` and `KeyIterator`, but it took a bit of thought.